File: app/src/main/java/com/unistay/android/domain/model/model/Room.kt
package com.unistay.android.domain.model.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "rooms")
data class Room(
    @PrimaryKey val roomId: String = "",
    val roomCode: String? = null,
    val title: String? = "",
    val address: String? = "",
    val district: String? = "",
    val basePrice: Double = 0.0,
    val targetUniversity: String? = "",
    val distanceToCampusKm: Double = 0.0,
    val status: String? = "Còn phòng",
    val maxOccupancy: Int = 1,
    val currentOccupancy: Int = 0,
    val amenities: String? = "",
    val imageUrl: String? = "",
    val contactPhone: String? = "",
    val description: String? = null,
    val roomType: String? = "Phòng trọ khép kín",
    @com.google.gson.annotations.SerializedName("userId")
    val ownerId: String? = "",
    val ownerName: String? = null
)

File: UniStay.Api/Models/Room.cs
using System.ComponentModel.DataAnnotations;

namespace UniStay.Api.Models
{
    public class Room
    {
        [Key]
        public string RoomId { get; set; } = string.Empty;
        public string RoomCode { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public decimal BasePrice { get; set; }
        public string TargetUniversity { get; set; } = string.Empty;
        public decimal DistanceToCampusKm { get; set; }
        public string Status { get; set; } = "Còn phòng";
        public int MaxOccupancy { get; set; }
        public int CurrentOccupancy { get; set; }
        public string? ImageUrl { get; set; }
        public string? Amenities { get; set; }
        public string? ContactPhone { get; set; }
        public string? Description { get; set; }
        public string RoomType { get; set; } = "Phòng trọ khép kín";
        public string UserId { get; set; } = string.Empty; // Landlord ID

        [System.ComponentModel.DataAnnotations.Schema.NotMapped]
        public string? OwnerName { get; set; }
    }
}
